src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -227,6 +220,58 @@ public virtual string AgentLicenseKey
             }
         }
 
+        private string TryGetLicenseKey()
+        {
+            // same order as old process - appsettings > env var(a/b) > newrelic.config
+            var candidateKeys = new Dictionary<string, string>
+            {
+                { "AppSettings", _configurationManagerStatic.GetAppSetting(Constants.AppSettingsLicenseKey) },
+                { "EnvironmentVariable", _environment.GetEnvironmentVariableFromList("NEW_RELIC_LICENSE_KEY", "NEWRELIC_LICENSEKEY") },
+                { "newrelic.config", _localConfiguration.service.licenseKey }
+            };
+
+            foreach (var candidateKey in candidateKeys)
+            {
+                // Did we find a key?
+                if (string.IsNullOrWhiteSpace(candidateKey.Value))
+                {
+                    continue;
+                }
+
+                // We found something, but is it a valid key?
+
+                // If the key is the default value from newrelic.config, we return the default value
+                // AgentManager.AssertAgentEnabled() relies on this behavior and will throw an exception if the key is the default value
+                if (candidateKey.Value.Equals("REPLACE_WITH_LICENSE_KEY"))
```

Comment:
A small suggestion: a lot of our sample code uses variations of `YOUR_LICENSE_KEY` for the environment variable. Maybe if we search for the string "license", it's a good bet that they haven't filled it in? That should cover us for all three key sources, and may be more helpful than the fall-through invalid key log messages.